ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/attestation/AggregatingAttestationPoolTest.java

```diff
@@ -400,21 +401,61 @@ public void getAttestations_shouldReturnAttestationsForGivenSlotOnly() {
         .containsExactly(attestation1);
   }
 
+  @Test
+  void onAttestationsIncludedInBlock_shouldNotAddAttestationsAlreadySeenInABlock() {
+    final AttestationData attestationData = dataStructureUtil.randomAttestationData(ZERO);
+    // Included in block before we see any attestations with this data
+    aggregatingPool.onAttestationsIncludedInBlock(
+        ONE, List.of(createAttestation(attestationData, 1, 2, 3, 4)));
+
+    // But still shouldn't be able to add a redundant attestation later
+    addAttestationFromValidators(attestationData, 2, 3);
+    assertThat(aggregatingPool.getSize()).isZero();
+  }
+
+  @Test
+  void onAttestationsIncludedInBlock_shouldRemoveAttestationsWhenSeenInABlock() {
+    final AttestationData attestationData = dataStructureUtil.randomAttestationData(ZERO);
+    addAttestationFromValidators(attestationData, 2, 3);
+
+    aggregatingPool.onAttestationsIncludedInBlock(
+        ONE, List.of(createAttestation(attestationData, 1, 2, 3, 4)));
+
+    assertThat(aggregatingPool.getSize()).isZero();
+  }
+
+  @Test
+  void onReorg_shouldBeAbleToReaddAttestations() {
```

Comment:
`void onReorg_shouldBeAbleToReadAttestations`

Just a typo.

Reply:
It actually does mean "re-add" as in add a second time in this case. I'm not sure if it's really a word but I find it very useful... :)